capstone/src/main/java/com/google/sps/data/RecommenderAlgorithm.java

```diff
@@ -0,0 +1,59 @@
+package com.google.sps.data;
+ 
+import com.google.sps.data.Room;
+import com.google.sps.data.User;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Entity;
+import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.Iterable;
+ 
+public class RecommenderAlgorithm {
+ 
+    public static ArrayList<Room> recommendRooms(User user) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        HashMap<String, Integer> userTagMap = RecommenderAlgorithm.populateUserTagMap(user);
+        
+        Query otherUserQuery = new Query(User.USER_ENTITY_NAME);
+        PreparedQuery otherUserResults = datastore.prepare(otherUserQuery);
+ 
+        User otherUser = null;
+        HashMap<String, Integer> otherUserTagMap = null;
+        HashMap<User, HashMap<String,Integer>> otherUserVectorMap = new HashMap<User, HashMap<String,Integer>>();
```

Comment:
Got it. I believe in a future PR, Kyle used embedded entities but we still need to cache the vector map when the user follows a room still. I think for now, the best approach is to leave this here since Kyle changed most of the foundation of the code and change his PR to cache the vector map when the user follows a room.